server/xpub-controller/manuscript.js

```diff
@@ -11,172 +11,57 @@ class Manuscript {
     this.storage = storage
     this.scienceBeamApi = scienceBeamApi
     this.pubsubManager = pubsubManager
+    this.filesHelper = new FilesHelper(this.config, this.scienceBeamApi)
+    this.manuscriptHelper = new ManuscriptHelper(
+      this.config,
+      this.userId,
+      this.storage,
+      this.filesHelper,
+    )
   }
 
   async upload(manuscriptId, file, fileSize) {
     const { ON_UPLOAD_PROGRESS } = this.pubsubManager.asyncIterators
 
-    if (fileSize > this.config.get('fileUpload.maxSizeMB') * 1e6) {
-      throw new Error(
-        `File size shouldn't exceed ${this.config.get(
-          'fileUpload.maxSizeMB',
-        )}MB`,
-      )
-    }
-    // ensure user can view manuscript
-    let manuscript = await ManuscriptModel.find(manuscriptId, this.userId)
+    this.filesHelper.validateFileSize(fileSize)
 
-    const { stream, filename, mimetype: mimeType } = await file
-    let fileEntity = await new FileModel({
-      manuscriptId,
-      url: `manuscripts/${manuscriptId}`,
-      filename,
-      type: 'MANUSCRIPT_SOURCE_PENDING',
-      mimeType,
-    }).save()
+    // ensure user can view manuscript
+    const manuscript = await ManuscriptModel.find(manuscriptId, this.userId)
 
-    const fileId = fileEntity.id
-
-    logger.info(
-      `Manuscript Upload Size: ${filename}, ${fileSize} | ${manuscriptId}`,
-    )
+    const fileData = await FilesHelper.generateFileEntity(file, manuscriptId)
```

Comment:
This should be called `fileEntity` rather than `fileData`, since the function we're calling is `generateFileEntity`.

Reply:
I looks like the functions returns back the data as well. It's worth renaming the function to `getFileData`

Reply:
`generateFileEntity` returns `{ fileEntity , stream }` not just the fileEntity